lib/datadog/profiling/agent_settings_resolver.rb

```diff
@@ -0,0 +1,340 @@
+# frozen_string_literal: true
+
+require 'uri'
+
+require_relative '../core/configuration/settings'
+require_relative '../core/configuration/agent_settings_resolver_common'
+require_relative '../tracing/configuration/ext'
+require_relative '../../ddtrace/transport/ext'
+
+module Datadog
+  module Profiling
+    # This class unifies all the different ways that users can configure how we talk to the agent.
+    #
+    # It has quite a lot of complexity, but this complexity just reflects the actual complexity we have around our
+    # configuration today. E.g., this is just all of the complexity regarding agent settings gathered together in a
+    # single place. As we deprecate more and more of the different ways that these things can be configured,
+    # this class will reflect that simplification as well.
+    #
+    # Whenever there is a conflict (different configurations are provided in different orders), it MUST warn the users
+    # about it and pick a value based on the following priority: code > environment variable > defaults.
+    #
+    # TODO: Decouple Tracing. Profiling relies on the deprecated_for_removal_transport_configuration_proc, which in turn
+    # relies on code and constants in the Tracing namespace. This class is an exact copy of
+    # Datadog::Tracing::Configuration::AgentSettingsResolver. When the proc is removed, this resolver may be replaced
+    # with the implementation in Core.
+    class AgentSettingsResolver
+      include Core::Configuration::AgentSettingsResolverCommon
+
+      AgentSettings = Class.new(BaseAgentSettings) do
```

Comment:
Tracing and Profiling override the BaseAgentSettings to include the `deprecated_for_removal_transport_configuration_proc`. This seemed like the least bad way to extend AgentSettings without forcing the version in Core to also have an attribute for the proc.